cfpb/cfgov-refresh
# Ask CFPB: Make question page category and breadcrumbs reflect referring page

When users come to an Ask question page from one of the new portal pages or from an Ask category page, the breadcrumbs and category shown on the page should reflect the referring page.

## Additions

- Check an Ask question page's referrer and determine the page's category and breadcrumbs based on result
- Limit the subcategories shown on an Ask question page to ones that belong to the page's category

## Testing

1. Check out branch
2. Navigate to a [question page with multiple categories](http://localhost:8000/ask-cfpb/what-are-my-rights-as-a-servicemember-under-the-equal-credit-opportunity-act-can-a-creditor-refuse-to-extend-credit-to-me-just-because-im-a-servicemember-en-2057). The breadcrumbs and subcategories shown on the page should be for the page's default (first) category -- in this case, `Auto Loans`.
3. Navigate to pages for Ask categories that also contain this question  ([Mortgages](http://localhost:8000/ask-cfpb/category-mortgages/) and [Bank accounts](http://localhost:8000/ask-cfpb/category-bank-accounts-and-services/)) and click the question's link (it's near the top of the first page of results for both categories). The breadcrumbs and subcategories shown should reflect the category of the referring page.
4. Open the [Bank accounts](http://localhost:8000/admin/pages/10196/edit/) consumer tools portal page in the Admin and update its slug to `bank-accounts`.  Add a link to the question (`en-2057`) on the page, publish the page, and follow the link. The breadcrumbs and subcategories should reflect the portal page & its related Ask category.

## Screenshots

### No referrer
<img width="1182" alt="screen shot 2017-08-02 at 8 20 45 am" src="https://user-images.githubusercontent.com/778171/28881038-b608a5c6-775b-11e7-991f-6a7ec7a148d5.png">


### Ask category page referrer
<img width="1181" alt="screen shot 2017-08-02 at 8 20 59 am" src="https://user-images.githubusercontent.com/778171/28881039-b8b451da-775b-11e7-8402-c0f34907913b.png">
<img width="1191" alt="screen shot 2017-08-02 at 8 21 09 am" src="https://user-images.githubusercontent.com/778171/28881040-ba7de1d4-775b-11e7-95c0-80a3c349a7a9.png">


### Portal page referrer
<img width="1186" alt="screen shot 2017-08-02 at 8 21 19 am" src="https://user-images.githubusercontent.com/778171/28881047-bed43256-775b-11e7-9612-35f6d9eb0f9e.png">


## Checklist
